cmd/pairbench/main.go

```diff
@@ -0,0 +1,129 @@
+// Copyright 2022 PairMesh, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/pairmesh/pairmesh/benchmark"
+	"github.com/pairmesh/pairmesh/benchmark/client"
+	"github.com/pairmesh/pairmesh/benchmark/config"
+	"github.com/pairmesh/pairmesh/benchmark/server"
+	"github.com/pairmesh/pairmesh/pkg/cmdutil"
+	"github.com/pairmesh/pairmesh/pkg/logutil"
+	"github.com/pairmesh/pairmesh/version"
+	"github.com/spf13/cobra"
+)
+
+func main() {
+	var (
+		role        string
+		mode        string
+		port        uint16
+		host        string
+		isBounceStr string
+		clients     uint16
+		payload     uint32
+		duration    uint16
+		examples    = cmdutil.Examples{
+			{
+				Example: "pairbench -r server -p 9736",
+				Comment: "Start pairbench in server role as an echo server, serving port 9736",
+			},
+			{
+				Example: "pairbench -r client -e 100.68.80.110 -p 9736 -c 12 -l 42 -d 60",
+				Comment: `Start pairbench in client role, connecting to echo server 100.68.80.110 port 9736, 
+spawning 12 clients, each request contains 42 bytes, and test for 60 seconds.`,
+			},
+		}
+	)
+	rootCmd := &cobra.Command{
+		Use: fmt.Sprintf("pairbench -r %s [flags]", cmdutil.Underline("<MODE>")),
+		Long: fmt.Sprintf(`pairbench starts with server or client role.
+
+- In server role, besides '-m %[1]s', additional parameters '-p %[2]s', '-b %[3]s' are optional.
+
+- In client role, besides '-m %[4]s', additional parameters '-e %[5]s', '-p %[2]s', '-c %[6]s', '-l %[7]s', '-d %[8]s' are optional.
+`,
+			cmdutil.Underline("server"),
+			cmdutil.Underline("<PORT>"),
+			cmdutil.Underline("<BOUNCE>"),
+			cmdutil.Underline("client"),
+			cmdutil.Underline("ENDPOINT"),
+			cmdutil.Underline("<CLIENTS>"),
+			cmdutil.Underline("<PAYLOAD>"),
+			cmdutil.Underline("<DURATION>"),
+		),
+		Example:       examples.String(),
+		Version:       version.NewVersion().String(),
+		SilenceUsage:  false,
+		SilenceErrors: false,
+		PreRun: func(cmd *cobra.Command, args []string) {
+			logutil.InitLogger()
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
```

Comment:
Do we need to check the user input parameters to avoid aggressive parameters to exhaust all computer resources?

e.g:
```shell
bin/pairbench -c 1000000000000
```